File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        Set<String> set = new HashSet();
        
        for(String[] clothe: clothes){
            if(!map.containsKey(clothe[1]))
                map.put(clothe[1],1);  
            else
                map.replace(clothe[1],map.get(clothe[1])+1);
            set.add(clothe[1]);
        }
        int cnt = 1;
        for(String clo: set){
            cnt *= (map.get(clo)+1);
        }
        return cnt -1; 
    }
}

File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    dic = {}
    for v,k in clothes:
        dic[k] = dic.get(k,0)+1
    
    cnt = list(map( (lambda x: x+1), dic.values()))
    result = 1
    for i in cnt:
        result *= i
    return result-1